### DISCUSSION TITLE
Tesla M3 Battery Level not showing

### DISCUSSION THREAD

**Discussion by Duff-Box1** [alandtse/tesla#489] [Q&A]
Hi I’ve just updated the Tesla custom integration which has an excellent set of information and control, but the battery % charge entity seems to have vanished from the dataset? Any suggestions welcome.
![C1923117-591D-4F99-A238-F1D89DBE8DBE](https://user-images.githubusercontent.com/56402252/213859166-1d49def2-a4d7-4039-9ed0-510783f1b402.png)

This error originated from a custom integration.

Logger: homeassistant
Source: custom_components/tesla_custom/sensor.py:406 
Integration: Tesla Custom Integration (documentation, issues) 
First occurred: 09:10:48 (576 occurrences) 
Last logged: 11:44:03

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 168, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 316, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 121, in async_update_listeners
    update_callback()
  File "/config/custom_components/tesla_custom/base.py", line 39, in refresh
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 556, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 597, in _async_write_ha_state
    state = self._stringify_state(available)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 562, in _stringify_state
    if (state := self.state) is None:
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 855, in state
    value = self.native_value
  File "/config/custom_components/tesla_custom/sensor.py", line 406, in native_value
    return round(self._energysite.solar_power)
TypeError: type NoneType doesn't define __round__ method

**Comment by tonywheeler60** [comment]
The entity that you need to add is called "sensor.NAMEOFCAR_battery" where NAMEOFCAR is whatever your car is identified as.  If you inspect any other of the already displayed entity values you'll see what needs to be used.  If you edit the existing card (that you've shown) you'll be able to add this by simply typing in "battery" - this will reveal a few and amngst them is what you need.  Sorry but I cannot help with the issue you've got - I am learning too!  Good luck.

**Comment by Duff-Box1** [comment]
Many thanks Tony, the problem I seem to have is that the entity is not present at all and searching battery does not find it unfortunately. It’s been working nicely for months and disappeared after I upgraded the integration and HA. Mystery.Sent from my iPhoneOn 21 Jan 2023, at 16:48, tonywheeler60 ***@***.***> wrote:﻿
The entity that you need to add is called "sensor.NAMEOFCAR_battery" where NAMEOFCAR is whatever your car is identified as.  If you inspect any other of the already displayed entity values you'll see what needs to be used.  If you edit the existing card (that you've shown) you'll be able to add this by simply typing in "battery" - this will reveal a few and amngst them is what you need.  Sorry but I cannot help with the issue you've got - I am learning too!  Good luck.

**Reply by Duff-Box1** [reply]
I found a fix for the disappearing battery %, the custom integration now seems to include home energy and I had both check boxes ticked. I already have a separate Tesla home energy integration so having it ticked in he new integration caused a conflict. I now have battery % again. Not sure if I should keep the separate home energy integration or delete and activate the one in the new custom integration. I will leave that for another day.
